plasmapy/simulation/tests/test_particle_tracker.py

```diff
@@ -0,0 +1,153 @@
+"""
+Tests for particle_tracker.py
+"""
+import astropy.units as u
+import numpy as np
+import pytest
+
+from hypothesis import given, settings
+from hypothesis import strategies as st
+
+from plasmapy.formulary.lengths import gyroradius
+from plasmapy.particles import CustomParticle
+from plasmapy.plasma.grids import CartesianGrid
+from plasmapy.simulation.particle_tracker import (
+    DiskIntervalSaveRoutine,
+    MemoryIntervalSaveRoutine,
+    NoParticlesOnGridsStoppingCondition,
+    ParticleTracker,
+    TimeElapsedStopCondition,
+)
+
+rng = np.random.default_rng()
+
+
+@pytest.mark.parametrize(
+    ("stop_condition", "save_routine", "is_disk_routine"),
+    [
+        (NoParticlesOnGridsStoppingCondition(), MemoryIntervalSaveRoutine, False),
+        (TimeElapsedStopCondition(10 * u.s), MemoryIntervalSaveRoutine, False),
+        (NoParticlesOnGridsStoppingCondition(), DiskIntervalSaveRoutine, True),
+        (TimeElapsedStopCondition(10 * u.s), DiskIntervalSaveRoutine, True),
+    ],
+)
+def test_interval_save_routine(tmp_path, stop_condition, save_routine, is_disk_routine):
```

Comment:
This isn't an ideal way to do this.. Would you have any recommendations @pheuer ? The problem is that I need a way of accessing `tmp_path` in order to instantiate a disk save routine, meaning the instantiation has to happen inside of the test function (unless there is another way of doing this?)

Reply:
Maybe something like `construct_save_routine_class` would clarify things (which you can pass `tmp_path` into)? Then we would just have to use a dictionary to retrieve the appropriate arguments

Reply:
Try pytest fixture
https://docs.pytest.org/en/6.2.x/fixture.html